synapse/handlers/oidc.py

```diff
@@ -247,6 +266,79 @@ async def handle_oidc_callback(self, request: SynapseRequest) -> None:
 
         await oidc_provider.handle_oidc_callback(request, session_data, code)
 
+    async def handle_backchannel_logout(self, request: SynapseRequest) -> None:
+        """Handle an incoming request to /_synapse/client/oidc/backchannel_logout
+
+        This extracts the logout_token from the request and tries to figure out
+        which OpenID Provider it is comming from. This works by matching the iss claim
+        with the issuer and the aud claim with the client_id.
+
+        Since at this point we don't know who signed the JWT, we can't just
+        decode it using authlib since it will always verifies the signature. We
+        have to decode it manually without validating the signature.
```

Comment:
Is it worth mentioning that `OidcProvider.handler_backchannel_logout` *does* then verify the token though?

Reply:
Probably, will do on Monday